server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/AllocationDecidersTests.java

```diff
@@ -53,18 +54,18 @@ public void testCheckAllDecidersBeforeReturningThrottle() {
         var allDecisions = generateDecisions(Decision.THROTTLE, () -> Decision.YES);
         var debugMode = randomFrom(RoutingAllocation.DebugMode.values());
         var expectedDecision = switch (debugMode) {
-            case OFF, EXCLUDE_YES_DECISIONS -> new Decision.Multi().add(Decision.THROTTLE);
+            case OFF -> Decision.THROTTLE;
+            case EXCLUDE_YES_DECISIONS -> new Decision.Multi().add(Decision.THROTTLE);
             case ON -> collectToMultiDecision(allDecisions);
         };
 
         verifyDecidersCall(debugMode, allDecisions, allDecisions.size(), expectedDecision);
     }
 
     public void testExitsAfterFirstNoDecision() {
-        var expectedDecision = Decision.NO;
-        var terminatingDecision = randomFrom(Decision.NO, Decision.single(Decision.Type.NO, "no with label", "explanation"));
-        var allDecisions = generateDecisions(terminatingDecision, () -> randomFrom(Decision.YES, Decision.THROTTLE));
-        var expectedCalls = allDecisions.indexOf(terminatingDecision) + 1;
+        var expectedDecision = randomFrom(Decision.NO, Decision.single(Decision.Type.NO, "no with label", "explanation"));
+        var allDecisions = generateDecisions(expectedDecision, () -> randomFrom(Decision.YES, Decision.THROTTLE));
+        var expectedCalls = allDecisions.indexOf(expectedDecision) + 1;
```

Comment:
Updated code bypasses actual decision with explanation (if any) to the caller code.
Please let me know if this should be reverted.

Reply:
should be fine 👍